File: resilient/resilient-annotation-processor/src/test/java/io/koraframework/resilient/annotation/processor/aop/testdata/CircuitBreakerTarget.java
package io.koraframework.resilient.annotation.processor.aop.testdata;

import io.koraframework.common.annotation.Component;
import io.koraframework.common.annotation.Root;
import io.koraframework.resilient.circuitbreaker.annotation.CircuitBreakable;

import java.io.IOException;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

@Component
@Root
public class CircuitBreakerTarget {

    public boolean alwaysFail = true;

    @CircuitBreakable(TestCircuitBreaker.class)
    public String getValueSync() {
        if (alwaysFail)
            throw new IllegalStateException("Failed");

        return "OK";
    }

    @CircuitBreakable(TestCircuitBreaker.class)
    public void getValueSyncVoid() {
        if (alwaysFail)
            throw new IllegalStateException("Failed");
    }

    @CircuitBreakable(TestCircuitBreaker.class)
    public void getValueSyncVoidCheckedException() throws IOException {
        if (alwaysFail)
            throw new IllegalStateException("Failed");
    }

    @CircuitBreakable(TestCircuitBreaker.class)
    public String getValueSyncCheckedException() throws IOException {
        if (alwaysFail)
            throw new IllegalStateException("Failed");

        return "OK";
    }

    @CircuitBreakable(TestCircuitBreaker.class)
    public CompletionStage<String> getValueStage() {
        if (alwaysFail)
            return CompletableFuture.failedFuture(new IllegalStateException("Failed"));

        return CompletableFuture.completedFuture("OK");
    }

    @CircuitBreakable(TestCircuitBreaker.class)
    public CompletableFuture<String> getValueFuture() {
        if (alwaysFail)
            return CompletableFuture.failedFuture(new IllegalStateException("Failed"));

        return CompletableFuture.completedFuture("OK");
    }
}

File: resilient/resilient-symbol-processor/src/test/kotlin/io/koraframework/resilient/symbol/processor/aop/testdata/CircuitBreakerTarget.kt
package io.koraframework.resilient.symbol.processor.aop.testdata

import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Root
import io.koraframework.resilient.circuitbreaker.annotation.CircuitBreaker
import io.koraframework.resilient.circuitbreaker.annotation.CircuitBreakable

@CircuitBreaker("custom1")
interface TestCircuitBreaker : io.koraframework.resilient.circuitbreaker.CircuitBreaker

@Component
@Root
open class CircuitBreakerTarget {

    var alwaysFail = true

    @CircuitBreakable(TestCircuitBreaker::class)
    open fun getValueSync(): String {
        check(!alwaysFail) { "Failed" }
        return "OK"
    }

    @CircuitBreakable(TestCircuitBreaker::class)
    open fun getValueSyncVoid() {
        check(!alwaysFail) { "Failed" }
    }

    // Throws here is an alias for kotlin.jvm.Throws
    // Method should compile normally
    @Throws(IllegalStateException::class)
    @CircuitBreakable(TestCircuitBreaker::class)
    open fun throws(): String {
        check(!alwaysFail) { "Failed" }
        return "OK"
    }
}
